pkg/operator/controller/controller.go

```diff
@@ -78,14 +76,15 @@ type reconciler struct {
 // Reconcile expects request to refer to a clusteringress in the operator
 // namespace, and will do all the work to ensure the clusteringress is in the
 // desired state.
-func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	result := reconcile.Result{}
+// Named return values are used to capture errors in deferred functions
+func (r *reconciler) Reconcile(request reconcile.Request) (result reconcile.Result, finalErr error) {
+	errs := []error{}
 
 	// TODO: Should this be another controller?
 	defer func() {
-		err := r.syncOperatorStatus()
-		if err != nil {
-			log.Error(err, "failed to sync operator status")
+		if err := r.syncOperatorStatus(); err != nil {
+			errs = append(errs, fmt.Errorf("failed to sync operator status: %v", err))
+			finalErr = utilerrors.NewAggregate(errs)
```

Comment:
This deferred function is becoming more entangled with the function's main logic.  Now that you have removed all but the final `return` in `Reconcile`, we could move the `syncOperatorStatus` into the main logic flow, which would simplify the error handling.